shared/appengine.go

```diff
@@ -121,17 +128,37 @@ func (a AppEngineAPIImpl) GetUploader(uploader string) (Uploader, error) {
 	return result, err
 }
 
-// GetHostname returns a cleaned-up hostname for the current environment.
+// GetHostname returns the canonical hostname for the current appengine project,
+// i.e. staging.wpt.fyi or wpt.fyi
 func (a AppEngineAPIImpl) GetHostname() string {
 	hostname := appengine.DefaultVersionHostname(a.ctx)
 	if hostname == "wptdashboard.appspot.com" {
 		return "wpt.fyi"
+	} else if hostname == "wptdashboard-staging.appspot.com" {
+		return "staging.wpt.fyi"
 	}
+	return hostname
+}
+
+// GetVersion returns the version name for the current environment.
+func (a AppEngineAPIImpl) GetVersion() string {
 	version := strings.Split(appengine.VersionID(a.ctx), ".")[0]
-	if version == "master" && hostname == "wptdashboard-staging.appspot.com" {
-		return "staging.wpt.fyi"
+	if appengine.IsDevAppServer() {
+		out, err := exec.Command("/usr/bin/git", "rev-parse", "--abbrev-ref", "HEAD").Output()
```

Comment:
Maybe don't do this... It'd shell out to `git` in a lot of requests and I don't see much benefit. Besides, even if we want to keep it, we should perhaps use `"git"` instead of the explicit absolute path as it'd be different on different platforms.

Reply:
1) only happens when running dev_appserver
1a) some requests use a version, but most are checks related so won't get called locally. The other is this use-case, which is nicer to see the git branch name as the version.
2) There's nothing on the path inside appengine, so can't use non-absolute
2b) failing to find git will just fall back to `dev_appserver`